### PR TITLE
Refactors embedding to use datums instead of storing data in bespoke elements

### PR DESCRIPTION
**Pull request by SmArtKar**
## About The Pull Request

This refactors embedding elements to make them use singleton datums (similarly to armor) instead being bespoke and creating a new element every time armor values are supposed to be adjusted.
Default values have been removed from defines due to now being declared in base class itself.
Additionally fixes vending machines and tackling gloves setting generated shards (which they instantly embed into their victim) embed properties to null after running the embedding code, despite said shards having non-null embedding values by default, making them not be able to embed into anyone else, also potentially breaking the pain/jostling code if they somehow get updated.

## Why It's Good For The Game

Current embedding system is an unnecessarily complicated mess as bespoke elements are hard to work with, and creating a new element every time you change values is hacky at best. This change should make it easier to read and work with.

## Changelog
:cl:
fix: Fixed glass shards generated from falling vending machines or tackling windows not being able to embed into anyone.
refactor: Refactored embedding code to use datums instead of bespoke elements and ugly associated lists.
/:cl:

### CODE REVIEW

File: code/game/objects/items/stacks/tape.dm [hunk 1]
```diff
@@ -43,7 +48,7 @@
 	if(!isitem(target))
 		return NONE
 
-	if(target.embedding && target.embedding == conferred_embed)
+	if(target.get_embed() && target.get_embed().type == conferred_embed)
```
Comment (MrMelbert) [inline code/game/objects/items/stacks/tape.dm:51]: `target.get_embed()?.type` ?

File: code/modules/surgery/bodyparts/_bodyparts.dm [hunk 2]
```diff
@@ -1096,22 +1096,15 @@
...
 /// Cleans up any attachment we have to the embedded object, removes it from our list
 /obj/item/bodypart/proc/_unembed_object(obj/item/unembed)
-	UnregisterSignal(unembed, COMSIG_ITEM_EMBEDDING_UPDATE)
 	embedded_objects -= unembed
 	refresh_bleed_rate()
 
-/obj/item/bodypart/proc/embedded_object_changed(obj/item/embedded_source)
-	SIGNAL_HANDLER
-	/// Embedded objects effect bleed rate, gotta refresh lads
-	refresh_bleed_rate()
```
Comment (MrMelbert) [inline code/modules/surgery/bodyparts/_bodyparts.dm:1113]: Did this code go anywhere?

Comment (SmArtKar) [reply code/modules/surgery/bodyparts/_bodyparts.dm]: forgot to push a second commit, oops

File: code/game/objects/items.dm [hunk 3]
```diff
@@ -1732,3 +1715,16 @@
...
+
+/// Fetches embedding data
+/obj/item/proc/get_embed()
+	RETURN_TYPE(/datum/embed_data)
+	return embed_type ? (embed_data ||= get_embed_by_type(embed_type)) : null
+
+/obj/item/proc/set_embed(datum/embed_data/embed)
+	if(embed_data == embed)
+		return
+	if(!(embed_data?.type in GLOB.embed_by_type))
```
Comment (MrMelbert) [inline code/game/objects/items.dm:1727]: Could/should be doing `!GLOB.embed_by_type[embed_data?.type]`, it'd be faster

**Comment by SmArtKar** [comment]
(I know that webedits are bad but this is literally a conflict from newlines, as proxy click PR also added lines to the end of the file)

**Pull request merged**